packages/plugin-wdm/src/device/device.js

```diff
@@ -194,9 +206,49 @@ const Device = SparkPlugin.extend({
       .then((res) => this._processRegistrationSuccess(res));
   },
 
+  @oneFlight
+  @waitForValue(`@`)
+  unregister() {
+    this.logger.info(`device: unregistering`);
+
+    if (!this.url) {
+      throw new Error(`device: not registered`);
+    }
+
+    return this.request({
+      uri: this.url,
+      method: `DELETE`
+    })
+      .then(() => this.clear());
+  },
+
   _processRegistrationSuccess(res) {
     this.logger.info(`device: received registration payload`);
     this.set(res.body);
+  },
+
+  _resetLogoutTimer() {
+    clearTimeout(this.logoutTimer);
+    this.unset(`logoutTimer`);
+    if (this.intranetInactivityCheckUrl && this.intranetInactivityDuration) {
+      this.on(`change:lastUserActivityDate`, () => this._resetLogoutTimer());
+
+      const timer = setTimeout(() => {
```

Comment:
If the user can move between networks and we don't know about it, your way is the way to go.